SerialPrograms/Source/PokemonSV/Programs/Farming/PokemonSV_MaterialFarmerTools.h

```diff
@@ -24,12 +24,12 @@ namespace PokemonAutomation{
 namespace NintendoSwitch{
 namespace PokemonSV{
 
-class MaterialFarmerOptions : public GroupOption{
+class MaterialFarmerOptions : public GroupOption, public ConfigOption::Listener{
 public:
+    ~MaterialFarmerOptions();
     MaterialFarmerOptions()
         : MaterialFarmerOptions(nullptr, nullptr, nullptr, nullptr, nullptr, nullptr)
```

Comment:
You don't need this. `std::unique_ptr` default initializes to null.

Reply:
Just do `MaterialFarmerOptions() = default`. (Which is the same thing as `MaterialFarmerOptions(){}`)

Reply:
Sorry for all the questions, but I still don't understand what you're suggesting. I tried changing the constructor on line 30 to `MaterialFarmerOptions() = default`, but then I get a compile error.  "function was implicitly deleted because 'PokemonAutomation::NintendoSwitch::PokemonSV::MaterialFarmerOptions' has an uninitialized data member"

Currently, `MaterialFarmerOptions()` delegates to the constructor with all the pointers, which is the reason for `MaterialFarmerOptions(nullptr, nullptr, nullptr, nullptr, nullptr, nullptr)`. How would setting it to `default` work in this case?

Reply:
Oh you're right. There's a bunch of other crap that needs to be initialized. Somehow I thought that other constructor was empty.

Nevermind then. It's fine as is.